src/analyses/base.ml

```diff
@@ -1998,18 +1998,14 @@ struct
     invalidate ~deep:true ~ctx (Analyses.ask_of_ctx ctx) gs st' deep_addrs
 
   let check_free_of_non_heap_mem ctx special_fn ptr =
-    match ctx.ask (Queries.MayPointTo ptr) with
-    | a when not (Queries.LS.is_top a) ->
-      let warn_if_not_heap_var special_fn var =
-        if not (ctx.ask (Queries.IsHeapVar var)) then
-          M.warn ~category:(Behavior (Undefined InvalidMemoryDeallocation)) ~tags:[CWE 590] "Free of non-dynamically allocated memory in function %s for pointer %a" special_fn.vname d_exp ptr
-      in
-      let pointed_to_vars =
-        Queries.LS.elements a
-        |> List.map fst
-      in
-      List.iter (warn_if_not_heap_var special_fn) pointed_to_vars
-    | _ -> ()
+    let points_to_set = ctx.ask (Queries.MayPointTo ptr) in
+    let exists_non_heap_var =
+      Queries.LS.elements points_to_set
```

Comment:
This will throw an exception for top, iirc. So one probably has to check for top separately.

Reply:
Yes, I completely forgot we might get `Unsupported` here :)
I added a `try ... with` now. (cf. https://github.com/goblint/analyzer/pull/1099/commits/ee80fd1761a7a780e46c2e9af25389bd268af65e)
Not sure if the `WARN` in the `with` of the `try` makes sense (when the points-to set is top), but decided to keep it there as a precaution, just in case.

Reply:
With the `try ... with`, `make test` now runs through again locally